Publicize: Apply legacy checks to not publicize in certain contexts

ebinnion opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Fixes an issue where bulk publishing posts would publicize many posts by disabling publicizing posts when in a bulk edit context.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

See p9MPsk-1BR-p2

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

- Checkout PR
- Ensure that you have active Publicize connections
- Publish a post
- Ensure that it gets publicized
- Create a draft post
- Go to your posts list
- Toggle the check box for your post
- Bulk edit
- Change the status to publish
- Save
- Ensure the post does not get published

spsiddarthan commented on projects/plugins/jetpack/modules/publicize/publicize.php:
```diff
@@ -1140,6 +1128,28 @@ public function save_meta( $post_id, $post ) {
+			return;
+		}
+
+		$submit_post = $this->should_submit_post_pre_checks( $post_id, $post );
```
Should this be `$submit_post = $this->should_submit_post_pre_checks( $post);` instead?

spsiddarthan commented on projects/packages/publicize/src/class-publicize-base.php:
```diff
@@ -1147,6 +1135,28 @@ public function save_meta( $post_id, $post ) {
+			return;
+		}
+
+		$submit_post = $this->should_submit_post_pre_checks( $post_id, $post );
```
Should this be `$submit_post = $this->should_submit_post_pre_checks( $post);`?

spsiddarthan reviewed (commented):
This tests well, but there are a few failing tests and I wonder if they might have to do with the params passed to the `should_submit_post_pre_checks` function. I have added a couple of inline comments about them.

Also, are we sure when we bulk edit a post and set the post status to publish from the draft, we shouldn't be publicizing them? That appears to be the goal of this PR, just wanted to confirm it once.

ebinnion commented:
Thank you for the review @spsiddarthan. You are correct that I was sending in the arguments incorrectly. I have since fixed that.

I have now added a Facebook connection and tested the above test cases as well as the case of not publicizing when bulk publishing.

> Also, are we sure when we bulk edit a post and set the post status to publish from the draft, we shouldn't be publicizing them? That appears to be the goal of this PR, just wanted to confirm it once.

For now, yes, that is the intention. Also, interestingly, it looks like we had already been doing this in the legacy versions of Publicize. I didn't write all of the cases for not publicizing myself. My guess is that we stopped applying that logic around the time we moved to the new version of sync, which was around version 4.2 of Jetpack.

spsiddarthan commented:
Thanks for the fixes and explanation on legacy publicize, @ebinnion! 

This tests well now!

coder-karen reviewed (approved):
Tests well for me too :) 

Left a couple of suggestions in terms of using `$$next-version$$` but I can also commit those as they're only related to the doc blocks.

coder-karen reviewed (approved):
Oops, in adding those doc suggestions means I've now dismissed all approvals. Adding mine back in again, @spsiddarthan not sure if you wanted to add another back in for Reach?
Edit - will go ahead and merge so this can make it into the next WoA release.

ebinnion commented:
r244994-wpcom

pablinos commented:
> r244994-wpcom

That looks to be another commit. This has been deployed in r245813-wpcom
